assets/js/frontend/course-video/video-blocks-extension.js

```diff
@@ -0,0 +1,121 @@
+( () => {
+	const {
+		courseVideoRequired,
+		courseVideoAutoComplete,
+		courseVideoAutoPause,
+	} = window.sensei.courseVideoSettings;
+
+	const preventClick = ( event ) => {
+		event.preventDefault();
+		return false;
+	};
+
+	function disableCompleteLessonButton() {
+		document
+			.querySelectorAll( '[data-id="complete-lesson-button"]' )
+			.forEach( ( button ) => {
+				button.disabled = true;
+				button.addEventListener( 'click', preventClick );
+			} );
+	}
+
+	const enableCompleteLessonButton = () => {
+		document
+			.querySelectorAll( '[data-id="complete-lesson-button"]' )
+			.forEach( ( button ) => {
+				button.removeEventListener( 'click', preventClick );
+				button.disabled = false;
+			} );
+	};
+
+	const submitCompleteLessonForm = () => {
+		const completeButton = document.querySelector(
+			'[data-id="complete-lesson-button"]'
+		);
+		if ( completeButton ) {
+			completeButton.click();
+		}
+	};
+
+	const onYouTubePlayerStateChange = ( event ) => {
+		const playerStatus = event.data;
+
+		if ( courseVideoRequired && playerStatus === YT.PlayerState.ENDED ) {
+			enableCompleteLessonButton();
+		}
+
+		if (
+			courseVideoAutoComplete &&
+			playerStatus === YT.PlayerState.ENDED
+		) {
+			submitCompleteLessonForm();
+		}
+	};
+
+	const initYouTubePlayer = ( iframe ) => {
+		const player = new YT.Player( iframe, {
+			events: {
+				onStateChange: onYouTubePlayerStateChange,
+			},
+		} );
+
+		if ( courseVideoAutoPause && document.hidden !== undefined ) {
+			// eslint-disable-next-line @wordpress/no-global-event-listener
+			document.addEventListener(
+				'visibilitychange',
+				() => {
+					if ( document.hidden ) {
+						player.pauseVideo();
+					}
+				},
+				false
+			);
+		}
+	};
+
+	// onYouTubeIframeAPIReady is called by YouTube iframe API when it is ready.
+	window.onYouTubeIframeAPIReady = () => {
+		document
+			.querySelectorAll(
+				'.sensei-course-video-container.youtube-extension iframe'
+			)
+			.forEach( initYouTubePlayer );
+	};
+
+	const onEnded = () => {
+		if ( courseVideoRequired ) {
+			enableCompleteLessonButton();
+		}
+
+		if ( courseVideoAutoComplete ) {
+			submitCompleteLessonForm();
+		}
+	};
+
+	const initVideoPlayer = ( video ) => {
+		video.addEventListener( 'ended', onEnded );
+
+		if ( courseVideoAutoPause && document.hidden !== undefined ) {
+			// eslint-disable-next-line @wordpress/no-global-event-listener
+			document.addEventListener(
+				'visibilitychange',
+				( video ) => () => {
```

Comment:
I think it will receive the event object instead of the video here. But since you already have the video in the scope, I don't think you need it here, right? I think it should be something very similar to the [other event](https://github.com/Automattic/sensei/pull/4562/files#diff-13f5badad7159f51a2c58277df50cacfc1aec7150c538a5aac9b1d25c9560edaR66), no?
Also, it didn't work for me, probably because it should just be one function.

Reply:
Thanks! Fixed and tested. Now it works.